snorkel/analysis/utils.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import scipy.sparse as sparse
+import torch
+
+from snorkel.types import ArrayLike
+
+
+def prob_to_pred(prob: np.ndarray) -> np.ndarray:
+    """Convert an array of probabilistic labels into an array of predictions
+
+    Parameters
+    ----------
+    prob
+        A [num_datapoints, num_classes] array of probabilistic labels such that each
+        row sums to 1.
+
+    Returns
+    -------
+    np.ndarray
+        A [num_datapoints, 1] array of predictions (integers in [1, ..., num_classes])
+    """
+    assert prob.ndim == 2
+    assert (prob <= 1).all()
+
+    pred = np.argmax(prob, axis=1) + 1
+
+    n, k = prob.shape
+    assert (pred >= 1).all()
+    assert (pred <= k).all()
+
+    return pred
+
+
+def pred_to_prob(pred: np.ndarray, num_classes: int) -> np.ndarray:
+    """Convert an array of probabilistic labels into an array of predictions
+
+    Parameters
+    ----------
+    pred
+        A [num_datapoints] or [num_datapoints], 1] array of predictions
+
+    Returns
+    -------
+    np.ndarray
+        A [num_datapoints, num_classes] array of probabilistic labels with probability
+        of 1.0 in the column corresponding to the prediction
+    """
+    pred = pred.reshape(-1, 1)
+    n = pred.shape[0]
+
+    prob = np.zeros((n, num_classes))
+
+    for idx, class_idx in enumerate(pred):
+        prob[idx, class_idx - 1] = 1.0
+
+    return prob
+
+
+def arraylike_to_numpy(array_like: ArrayLike) -> np.ndarray:
+    """Convert a 1d array-like (e.g,. list, tensor, etc.) to an np.ndarray"""
+
+    orig_type = type(array_like)
+
+    # Convert to np.ndarray
+    if isinstance(array_like, np.ndarray):
+        pass
+    elif isinstance(array_like, list):
+        array_like = np.array(array_like)
+    elif isinstance(array_like, sparse.spmatrix):
+        array_like = array_like.toarray()
+    elif isinstance(array_like, torch.Tensor):
+        array_like = array_like.numpy()
+    elif not isinstance(array_like, np.ndarray):
+        array_like = np.array(array_like)
+    else:
+        msg = f"Input of type {orig_type} could not be converted to 1d " "np.ndarray"
+        raise ValueError(msg)
+
+    # Correct shape
+    if (array_like.ndim > 1) and (1 in array_like.shape):
```

Comment:
If we just use squeeze, won't this work without the check?

Reply:
Addressed in a subsequent PR in the pipeline.